redis/conn.go

```diff
@@ -137,6 +148,21 @@ func Dial(network, address string, options ...DialOption) (Conn, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if do.dialTLS {
+		tlsConfig := do.tlsConfig
+		if tlsConfig == nil {
+			// https://golang.org/pkg/crypto/tls/#Client
+			// At this point we don't know the ServerName.
+			tlsConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
And nope, that doesn't work because I won't be able to override things w/o doing this:

```go
c, err := DialURL("rediss://localhost:1234", DialNetDial(func(network, address string) (net.Conn, error) {
  tls.Dial(network, address, &tls.Config{InsecureSkipVerify: true})
})
```

Which is a lot more verbose and then what's the point in all of this.